packages/arb-validator-core/gobridge/inboxTopChallenge.go

```diff
@@ -88,12 +91,14 @@ func (c *inboxTopChallenge) OneStepProof(
 	lowerHashA common.Hash,
 	value common.Hash,
 ) error {
+	c.client.goEthMutex.Lock()
+	defer c.client.goEthMutex.Unlock()
 	matchHash := valprotocol.InboxTopChallengeDataHash(lowerHashA, valprotocol.AddMessageToPending(lowerHashA, value), big.NewInt(1))
-	if !c.client.challenges[c.contractAddress].challengerDataHash.Equals(matchHash) {
+	if !c.client.challenges[c.contractAddress].challengeData.challengerDataHash.Equals(matchHash) {
```

Comment:
Unnecessary indirection to get `challengeData`

Reply:
fixed